core/src/main/java/org/eobjects/datacleaner/extension/output/CreateCsvFileAnalyzer.java

```diff
@@ -72,13 +83,22 @@
     @Configured(order = 5, required = false)
     boolean includeHeader = true;
 
+    @Configured(order = 6, required = false)
+    InputColumn<?> columnToBeSortedOn ;
+
     @Inject
     @Provided
     UserPreferences userPreferences;
 
+    private File _tempFile;
+
+    @Initialize
+    public void initTempFile() throws Exception {
+        _tempFile = File.createTempFile("csv_file_analyzer", ".csv");
```

Comment:
I guess that if "columnToBeSortedOn" is null (sorting is turned off), we should not write to a temp file. We should write to the final file directly. So maybe let's rename the "_tempFile" variable and call it something a la "_targetFile". And initialize it according to the null check.